src/main/webapp/components/TopNavigationMenu.js

```diff
@@ -0,0 +1,260 @@
+    if (user != null && user.data.session) {
+      let resp = await User.getUserSession();
+      Storage.setCurrentUser(resp.data);
+      if (this._isMounted) {
+        this.setState({
+          isLogged: true,
+          userSession: resp.data
+        });
+      }
+    }
+  }
+
+  onSuccess = async(token) => {
+    await User.signIn(token);
+    Storage.setUserIsLogged(true);
+    this.init().then(resp => {
+      if (Storage.getLocationFrom() != null) {
+        this.props.history.push(Storage.getLocationFrom());
+        Storage.removeLocationFrom();
+      } else {
+        this.props.history.push("/index");
+      }
+    });
+  };
+
+  signOut = async() => {
+    this.props.showSpinner();
+    Storage.clearStorage();
+    if (window.gapi.auth2 != undefined) {
+      let auth2 = window.gapi.auth2.getAuthInstance();
+      await auth2.signOut();
+      await User.signOut();
+    }
+    this.setState({
+      isLogged: false
+    }, () => this.props.hideSpinner());
+  };
+
+  openMetricsReport() {
+    Reports.getMetricsReport();
+  }
+
+  loadOptions(query, callback) {
+    let options = [];
+    if (query.length > 2) {
+      Search.getMatchingIssues(query).then(response => {
+        let options = response.data.map(function (item) {
+          let label = item.label;
+          if (item.linkDisabled === true && item.pm.length > 0) {
+            label =
+              span({ style: styles.listResultContainer }, [
+                span({ style: styles.badgeContactAccess }, [
+                  'Please contact ' + item.pm + ' for access'
+                ]),
+                div({ style: { 'marginTop': '10px' } }, [
+                  item.label
+                ])
+              ])
+          } else if (item.linkDisabled === true) {
+            label =
+              span({ style: styles.listResultContainer }, [
+                span({ style: styles.badgeContactAccess }, [
+                  ' Please contact ' + item.reporter + ' for access'
+                ]),
+                div({ style: { 'marginTop': '10px' } }, [
+                  item.label
+                ])
+              ])
+          }
+          return {
+            key: item.id,
+            value: item.value,
+            label: label,
+            linkDisabled: item.linkDisabled,
+            url: item.url,
+            isDisabled: item.linkDisabled
+          };
+        });
+        callback(options);
+      }).catch(e => {
+        callback(options)
+      });
+    } else {
+      callback(options)
+    }
+  };
+
+  handleSearchChange = (data, action) => {
+    if (data != null && !data.linkDisabled) {
+      window.location.href = data.url;
+    }
+  };
+
+  isBroadUser() {
+    return this.state.isLogged && this.state.userSession != null &&
+      this.state.userSession.isBroad;
+  }
+
+  isAdmin() {
+    return this.state.isLogged && this.state.userSession != null &&
+      (this.state.userSession.isAdmin || this.state.userSession.isORSP);
+  }
```

Comment:
This seems like an alternate pattern for determining if the user is an admin or not. Why not use `component.isAdmin` and make any necessary changes there so they are consistent through out the application? Same thing applies to `isViewer` and `isBroadUser`.